backend/tracim_backend/__init__.py

```diff
@@ -131,11 +131,13 @@ def web(global_config: OrderedDict, **local_settings) -> Router:
     app_config.configure_filedepot()
     settings["CFG"] = app_config
 
+    configurator = Configurator(settings=settings, autocommit=True)
+    configurator.begin()
+
     # Init plugin manager
     plugin_manager = init_plugin_manager(app_config)
-    settings["plugin_manager"] = plugin_manager
+    configurator.registry.settings["plugin_manager"] = plugin_manager
```

Comment:
This creates the `Configurator` *before* loading the plugins, making the `Registry` and, therefore, the `Configurator` accessible on the `threadlocal` stack for plugins to register routes, views, modify settings...

`configurator.begin()` pushes the `Registry` on the `threadlocal` stack. It is later popped with `configurator.end()`

`Registry` and `Configurator` can be accessed in this manner within a plugin's `on_plugins_loaded` hook.

```python
    @hookimpl
    def on_plugins_loaded(self, plugin_manager: PluginManager) -> None:
        """
        This method is called when the plugin is loaded.
        """
        from pyramid.threadlocal import get_current_registry
        registry = get_current_registry()
        config = Configurator(registry=registry)
        config.add_route("my_route", "/plugins/my_route", request_method="GET")
        config.add_view(self.my_view, route_name="my_route")
 ```

Reply:
Since the hook is called before starting the server and (at least I think so) the whole operation is single threaded, this is safe for use and follows standard usage of Pyramid's `Configurator`

Reply:
> (at least I think so) [...] this is safe for use

Could you check this information and put it in the example?

You should also:
- put a link to this documentation in the code example
- move the example here to the #6235 issue description in order to make information available for future readers (nobody reads PR comments)

> and follows standard usage of Pyramid's Configurator

Could you link to the associated documentation? (in the code and in the issue description, not in the comments here)

Reply:
I will put the required information and links to the documentation in the issue and the code